src/Features/CSharp/Portable/InvertIf/CSharpInvertIfCodeRefactoringProvider.cs

```diff
@@ -193,6 +198,35 @@ protected override StatementSyntax AsEmbeddedStatement(IEnumerable<StatementSynt
                 : updatedIf.WithAdditionalAnnotations(Formatter.Annotation);
         }
 
+        private static bool ShouleKeepFalse(IfStatementSyntax ifStatement, StatementSyntax falseStatement)
+        {
+            if (falseStatement is BlockSyntax falseBlock)
+            {
+                // Block false syntax with some statements included.
+                // If there are no statements, it's an empty
+                // block and we can get rid of it.
+                if (falseBlock.Statements.Any())
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                // The statement is not expected to have children, so we know it's fine
+                // to consider this something that needs to be included. Such as 
+                // a return statement, or other similar things for single line if/else.
+                return true;
+            }
+
+            // If the stateements for the else don't pass, we still need to check
+            // if there are comments from the original if that should be included.
+            // If so, pass them along to be copied to the new else block
+            return ifStatement.Statement is BlockSyntax block
```

Comment:
It would probably make more sense to put this in the if block above. Its a little confusing coming after an `else { return; }`. I would say "you should invert the if", but that seems a little too ironic to be in good taste :)

Reply:
it's a bit confusing, because it's checking different things. One checks the original IfStatementSyntax , the other checks the else StatementSyntax

Reply:
I renamed to "originalIfStatement" to clarify

Reply:
I just meant you could do:

```cs
if (falseStatement...)
{
	if (falseBlock.Statements.Any())
	{
	}

	return ifStatement.blalah
}

return true;
```

Reply:
ah gotcha. Yea I like that, more clear.